src/query_ctrl.ts

```diff
@@ -24,8 +24,8 @@ export class GenericDatasourceQueryCtrl extends QueryCtrl {
     this.showJSON = false;
   }
 
-  getOptions(query) {
-    return this.datasource.metricFindQuery(query || '');
+  findMetrics(query: string) {
+    return this.datasource.findMetricsQuery(this.target.type, query);
```

Comment:
ah, sorry. We had it (`|| ''`) handled here. I'll try to test it and handle it properly after we merge this.

Reply:
I don't fancy conditions like `if(query)` when query is not boolean. We should check for `undefined` or `null` explicitly.